src/main/java/org/wildfly/security/pem/Pem.java

```diff
@@ -163,6 +177,19 @@ private static X509Certificate parsePemX509CertificateContent(String type, ByteI
         }
     }
 
+    private static PublicKey parsePemPublicKey(String type, ByteIterator content) throws IllegalArgumentException {
+        if (! type.equals("PUBLIC KEY")) {
+            throw log.invalidPemType("PUBLIC KEY", type);
+        }
+        try {
+            X509EncodedKeySpec spec = new X509EncodedKeySpec(content.drain());
+            KeyFactory kf = KeyFactory.getInstance("RSA");
```

Comment:
What if the key isn't an RSA key?

The structure of X.509 key specifications is:
```
 SubjectPublicKeyInfo ::= SEQUENCE {
   algorithm AlgorithmIdentifier,
   subjectPublicKey BIT STRING }
```
It should be possible to use the DER decoder to read the algorithm OID, decode it to a JCA algorithm name, and then create an algorithm-specific keyspec and key from that using the appropriate KeyFactory.